test/SlowTests/Issues/RavenDB_14939.cs

```diff
@@ -0,0 +1,252 @@
+                Assert.Contains("Cannot find analyzer type 'MyAnalyzer' for field: Name", e.Message);
+
+                store.Maintenance.Send(new PutAnalyzersOperation(new AnalyzerDefinition
+                {
+                    Name = "MyAnalyzer",
+                    Code = GetAnalyzer("RavenDB_14939.MyAnalyzer.cs")
+                }));
+
+                store.ExecuteIndex(new MyIndex());
+
+                Fill(store);
+
+                WaitForIndexing(store);
+
+                AssertCount<MyIndex>(store);
+
+                store.Maintenance.Send(new DeleteAnalyzerOperation("MyAnalyzer"));
+
+                store.Maintenance.Send(new ResetIndexOperation(new MyIndex().IndexName));
+
+                var errors = WaitForIndexingErrors(store);
+                Assert.Equal(1, errors.Length);
+                Assert.Equal(1, errors[0].Errors.Length);
+                Assert.Contains("Cannot find analyzer type 'MyAnalyzer' for field: Name", errors[0].Errors[0].Error);
+            }
+        }
+
+        [Fact]
+        public void CanUseCustomAnalyzerWithConfiguration()
+        {
+            using (var store = GetDocumentStore(new Options
+            {
+                ModifyDatabaseRecord = record =>
+                {
+                    record.Analyzers = new Dictionary<string, AnalyzerDefinition>
+                    {
+                        { "MyAnalyzer", new AnalyzerDefinition
+                        {
+                            Name = "MyAnalyzer",
+                            Code = GetAnalyzer("RavenDB_14939.MyAnalyzer.cs")
+                        }}
+                    };
+
+                    record.Settings[RavenConfiguration.GetKey(x => x.Indexing.DefaultSearchAnalyzer)] = "MyAnalyzer";
+                }
+            }))
+            {
+                store.ExecuteIndex(new MyIndex_WithoutAnalyzer());
+
+                Fill(store);
+
+                WaitForIndexing(store);
+
+                AssertCount<MyIndex_WithoutAnalyzer>(store);
+            }
+        }
+
+        [Fact]
+        public void CanUseCustomAnalyzerWithConfiguration_NoAnalyzer()
+        {
+            using (var store = GetDocumentStore(new Options
+            {
+                ModifyDatabaseRecord = record =>
+                {
+                    record.Settings[RavenConfiguration.GetKey(x => x.Indexing.DefaultSearchAnalyzer)] = "MyAnalyzer";
+                }
+            }))
+            {
+                store.ExecuteIndex(new MyIndex_WithoutAnalyzer());
+
+                Fill(store);
+
+                var errors = WaitForIndexingErrors(store);
+                Assert.Equal(1, errors.Length);
+                Assert.Equal(1, errors[0].Errors.Length);
+                Assert.Contains("Cannot find analyzer type 'MyAnalyzer' for field: @default", errors[0].Errors[0].Error);
+
+                store.Maintenance.Send(new PutAnalyzersOperation(new AnalyzerDefinition
+                {
+                    Name = "MyAnalyzer",
```

Comment:
If I changed to to `MyFooBar`, how would this impact other things?

Reply:
New analyzer registered under `MyFooBar`